### PR TITLE
Project structure

### PR DESCRIPTION
**Pull request by xancoder**
This is the pull request referencing bug #24. Please refer to the bug to find the more detailed description of what I did. 

This code worked fine on my system, but I would very much appreciate it if you could test it on yours as well, just to make sure that it works on all sorts of setups.

Please don't hesitate to contact me if you have any questions or things you'd like me to add/change.

### CODE REVIEW

File: initramfs/init-premount [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+fi
+
+if [ -z "${ALLOWED_IPS}" ]; then
+  log_failure_msg 'Allowed IPs is not defined'
+  return 1
+fi
+log_end_msg
+
+log_begin_msg 'set local date time'
+date_from_web=$(wget -qSO- "${DATETIME_URL}" 2>&1 | sed -n 's/^ *Date: *//p' | head -n 1)
```
Comment (r-pufky) [inline initramfs/init-premount]: This assumes wget is installed by default on debian and Pi installations. Maybe these should be installed as part of the build process?

File: scripts/migrate_project_structure.sh [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+PEER_URL=${tmo_PEER_URL}
+PEER_ENDPOINT="\${PEER_URL}:\${PEER_PORT}"
+
+# Persistent Keepalive. Required to ensure connection for non-exposed ports.
+PERSISTENT_KEEPALIVES=25
+
+# Allowed IP addresses (CIDR) on wireguard; for boot this should be the peer (server).
+ALLOWED_IPS=172.31.255.254/32
+
+# url to send a web request to set the local datetime
```
Comment (r-pufky) [inline scripts/migrate_project_structure.sh]: Include explicit instructions on disabling external URL query, e.g. leaving this blank.

**Review by r-pufky — CHANGES_REQUESTED** [review]
Two comments pertaining to assumption of installed utilities and explicit instructions for disabling external URL queries. Great job - thanks for all the work!

File: initramfs/init-premount (hunk 1 above)
Comment (xancoder) [reply initramfs/init-premount]: Thank you for drawing my attention to this! I fixed it by adding it to the install dependencies in the Makefile.

File: scripts/migrate_project_structure.sh (hunk 2 above)
Comment (xancoder) [reply scripts/migrate_project_structure.sh]: The DATETIME_URL is now optional. If it is left blank, this step is skipped.

**Comment by r-pufky** [comment]
Thanks for the changes!

**Pull request merged**